libs/src/api/account.js

```diff
@@ -100,91 +100,58 @@ class Account extends Base {
-          metadata = await this.User.uploadProfileImages(profilePictureFile, coverPhotoFile, metadata)
-
-          phase = phases.ADD_CREATOR
-          userId = await this.User.addCreator(metadata)
-        }
+      this.REQUIRES(Services.CREATOR_NODE)
+
+      if (this.web3Manager.web3IsExternal()) {
+        this.REQUIRES(Services.IDENTITY_SERVICE)
+
+        phase = phases.CREATE_USER_RECORD
+        await this.identityService.createUserRecord(email, this.web3Manager.getWalletAddress())
       } else {
-        if (this.web3Manager.web3IsExternal()) {
-          // Non-creator and external web3 (e.g. MetaMask)
-          this.REQUIRES(Services.IDENTITY_SERVICE)
-
-          phase = phases.UPLOAD_PROFILE_IMAGES
-          metadata = await this.User.uploadProfileImages(profilePictureFile, coverPhotoFile, metadata)
-
-          phase = phases.ADD_USER
-          userId = await this.User.addUser(metadata)
-
-          phase = phases.CREATE_USER_RECORD
-          await this.identityService.createUserRecord(email, this.web3Manager.getWalletAddress())
-        } else {
-          // Non-creator and identity service web3
-          this.REQUIRES(Services.IDENTITY_SERVICE, Services.HEDGEHOG)
-
-          // If an owner wallet already exists, don't try to recreate it
-          if (!hasWallet) {
-            phase = phases.HEDGEHOG_SIGNUP
-            const ownerWallet = await this.hedgehog.signUp(email, password)
-            await this.web3Manager.setOwnerWallet(ownerWallet)
-            await this.generateRecoveryLink({ handle: metadata.handle, host })
-          }
-
-          phase = phases.UPLOAD_PROFILE_IMAGES
-          metadata = await this.User.uploadProfileImages(profilePictureFile, coverPhotoFile, metadata)
-
-          phase = phases.ADD_USER
-          userId = await this.User.addUser(metadata)
+        this.REQUIRES(Services.HEDGEHOG)
+        // If an owner wallet already exists, don't try to recreate it
+        if (!hasWallet) {
+          phase = phases.HEDGEHOG_SIGNUP
+          const ownerWallet = await this.hedgehog.signUp(email, password)
+          await this.web3Manager.setOwnerWallet(ownerWallet)
+          await this.generateRecoveryLink({ handle: metadata.handle, host })
         }
       }
-    } catch (err) {
-      return { error: err.message, phase }
-    }
 
-    metadata.user_id = userId
-    metadata.wallet = this.web3Manager.getWalletAddress()
-    this.userStateManager.setCurrentUser(metadata)
+      // Add user to chain
+      phase = phases.ADD_USER
+      userId = await this.User.addUser(metadata)
+
+      // Assign replica set to user, updates creator_node_endpoint on chain, and then update metadata object on content node + chain (in this order)
+      phase = phases.ADD_REPLICA_SET
+      metadata = await this.User.assignReplicaSet({
+        serviceProvider,
+        userId
+      })
+
+      // Upload profile pic to primary Content Node and sync across secondaries
```

Comment:
comment should be profile pic + cover photo

Reply:
ehhh technically `coverPhotoFile` will always be null because you cant upload a cover photo at signup

Reply:
That's not necessarily true -- it's only the case for *our* client. But this package allows you to submit both, so we should document that correctly